src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1562,13 +1562,18 @@ def process_remove_identity_namespace(cmd, namespace):
 
 
 def process_gallery_image_version_namespace(cmd, namespace):
-    TargetRegion = cmd.get_models('TargetRegion')
+    TargetRegion, EncryptionImages, OSDiskImageEncryption, DataDiskImageEncryption = cmd.get_models(
+        'TargetRegion', 'EncryptionImages', 'OSDiskImageEncryption', 'DataDiskImageEncryption')
     storage_account_types_list = [item.lower() for item in ['Standard_LRS', 'Standard_ZRS', 'Premium_LRS']]
     storage_account_types_str = ", ".join(storage_account_types_list)
 
     if namespace.target_regions:
+        if hasattr(namespace, 'target_region_encryption') and namespace.target_region_encryption:
```

Comment:
is it possible that namespace has no `target_region_encryption` attribute ? maybe we can remove `hasattr(namespace, 'target_region_encryption')` check.

Reply:
This function is invoked in multiple places. It is possible. Cannot remove the check.